router/batchrouter/asyncdestinationmanager/bing-ads/bulk_uploader.go

```diff
@@ -280,25 +236,36 @@ func (b *BingAdsBulkUploader) GetUploadStats(UploadStatsInput common.FetchUpload
 		initialEventList = append(initialEventList, job.JobID)
 	}
 	var eventStatsResponse common.GetUploadStatsResponse
-	filePaths := common.GenerateArrayOfStrings(UploadStatsInput.OutputFilePath)
-	for _, filePath := range filePaths {
-		response, _ := b.GetUploadStatsOfSingleImport(filePath)
-		eventStatsResponse = common.GetUploadStatsResponse{
-			Status: response.Status,
-			Metadata: common.EventStatMeta{
-				FailedKeys:    append(eventStatsResponse.Metadata.FailedKeys, response.Metadata.FailedKeys...),
-				ErrFailed:     nil,
-				WarningKeys:   []int64{},
-				ErrWarning:    nil,
-				SucceededKeys: []int64{},
-				ErrSuccess:    nil,
-				FailedReasons: common.MergeMaps(eventStatsResponse.Metadata.FailedReasons, response.Metadata.FailedReasons),
-			},
+	var failedJobIds []int64
+	var cumulativeFailedReasons map[string]string
+	var status string
+	fileURLs := common.GenerateArrayOfStrings(UploadStatsInput.PollResultFileURLs)
+	for _, fileURL := range fileURLs {
+		filePaths, err := b.DownloadAndGetUploadStatusFile(fileURL)
+		if err != nil {
+			return common.GetUploadStatsResponse{
+				Status: "500",
+			}
 		}
+		response, err := b.GetUploadStatsOfSingleImport(filePaths[0]) // only one file should be there
+		if err != nil {
+			return common.GetUploadStatsResponse{
```

Comment:
should be capture error in the response or should we log the error?

Reply:
same as above